src/MatomoTracker.js

```diff
@@ -91,13 +91,15 @@ class MatomoTracker {
    *
    * {String} `value` - The event value. Must be a float or integer value (numeric), not a string.
    *
+   * {String} `campaign` - The event related campaign.
+   *
    * {Object} `userInfo` - Optional data used for tracking different user info, see https://developer.matomo.org/api-reference/tracking-api#optional-user-info.
    */
-  trackEvent({ category, action, name, value, userInfo = {} }) {
+  trackEvent({ category, action, name, value, campaign, userInfo = {} }) {
     if (!category) throw new Error('Error: category is required.');
     if (!action) throw new Error('Error: action is required.');
 
-    return this.track({ e_c: category, e_a: action, e_n: name, e_v: value, ...userInfo });
+    return this.track({ e_c: category, e_a: action, e_n: name, e_v: value, mtm_campaign: campaign, ...userInfo });
```

Comment:
Hey, thanks for requesting changes. How do you think about passing `mtm_campaign` through `userInfo` object, which is spread in the tracking method?
This means that `trackEvent({ category, action, name, value, userInfo: { mtm_campaign: campaign } })` with certain data results in the same as `trackEvent({ category, action, name, value, campaign })` in your code changes.

As described in the method comment, `userInfo` are "_optional data used for tracking different user info_". You can pass any Matomo params, that you want to have been tracked.

I can't say that this code change adds new possibilities.
Can you please explain, how the code can benefit from your proposed changes?

Thanks again and have a nice day.

Reply:
Hi @donni106 and thank you for replying so quickly! 

I must apologies for being so slow in noticing your answer. For some obscure reasons, I did not receive the notification for your reply...

Anyways 🙂 here is what I think

> How do you think about passing mtm_campaign through userInfo object, which is spread in the tracking method?
This means that trackEvent({ category, action, name, value, userInfo: { mtm_campaign: campaign } }) with certain data results in the same as trackEvent({ category, action, name, value, campaign }) in your code changes.

Actually I cannot do that as I use typescript and the `userInfo` property in `Event` type does not contain `mtm_campaign` only `uid`...
<img width="1221" alt="Screenshot 2023-12-19 at 15 20 01" src="https://github.com/donni106/matomo-tracker-react-native/assets/9129812/37207bc1-815e-45f5-9335-c5a99655c61f">

> As described in the method comment, userInfo are "optional data used for tracking different user info". You can pass any Matomo params, that you want to have been tracked.

Indeed, although it also says `..., see https://developer.matomo.org/api-reference/tracking-api#optional-user-info.` and looking closely at the provided documentation there is no mention of `mtm_campaign` parameter, it is in [another section](https://developer.matomo.org/api-reference/tracking-api#optional-acquisition-channel-attribution)

> I can't say that this code change adds new possibilities.
Can you please explain, how the code can benefit from your proposed changes?

Yes the extra new parameter allows the tracking of campaign with the only required parameter `mtm_campaign` it is compulsory in case you are doing advertisement because you need to know how much the ads are bringing back compare to you investment. For instance if your ad cost 1k and the sells thanks to it are 1.3k your ad is profitable...

Reply:
Hi @donni106 I reply this in the main PR thread:

Actually I cannot do that as I use typescript and the `userInfo` property in `Event` type does not contain `mtm_campaign` only `uid`...
<img width="1221" alt="Screenshot 2023-12-19 at 15 20 01" src="https://github.com/donni106/matomo-tracker-react-native/assets/9129812/37207bc1-815e-45f5-9335-c5a99655c61f">

Reply:
Hello @donni106 how are you coding?

Can you please give me an update about this feature, can we eventually merge it to the code base or d I need to modify anything before it can be merged?

Regards
73k05

PS: I'll be in Berlin soon if you wanna catch up ^^

Reply:
thanks for reminding here. as i can understand, with the current typescript difficulties, we need this update. i will merge and release a new version.

Reply:
Thank you so much!! If you wanna meet in Berlin drop me an email [here](mailto:olivier@inkOS.fr)